src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -418,10 +426,12 @@ string IDurableOrchestrationContext.StartNewOrchestration(string functionName, o
                     System.Diagnostics.Debug.Assert(!oneWay, "The oneWay parameter should not be used for activity functions.");
                     if (retryOptions == null)
                     {
+                        this.IncrementActionsOrThrowException();
```

Comment:
It might be less maintenance to just call this once before the `switch` statement. Maybe after the call to `ThrowIfInvalidAccess()`?